### PR TITLE
feat: add extra annotations pgbouncer

### PR DESCRIPTION
**Pull request by dim-ops**
Add extra annotations for pgbouncer deployment

### CODE REVIEW

**Comment by dim-ops** [comment]
Hello @josh-fell, I'm not sure to understand why `Tests / Static checks`fails, can you explain me ?

**Comment by dim-ops** [comment]
I think I finally figured out the error in the IC `E501 Line too long` means I have too many characters on one line, I formatted my code another way.

I have another solution otherwise @josh-fell

**Comment by dim-ops** [comment]
@joshfell is it possible to merge ?

**Comment by josh-fell** [comment]
> @JoshFell is it possible to merge ?

I wouldn't be the best person to review this PR as I have never worked with the charts. (I just noticed there seemed to be very similar PRs open between yours and another.) I'd feel more comfortable if the code owners weighed in. 

I did kick off CI for you which was waiting for approval to run.

File: chart/values.yaml
Comment (jedcunningham) [inline chart/values.yaml]:
```suggestion
  podAnnotations: {}
```

Let's use podAnnotations instead for consistency across components.

File: tests/charts/test_pgbouncer.py
Comment (jedcunningham) [inline tests/charts/test_pgbouncer.py]: Can you add coverage [here](https://github.com/apache/airflow/blob/0fe73c2e8c0ca90faf54b39139ffab5769f4cca1/tests/charts/test_annotations.py#L371) instead?

File: chart/templates/pgbouncer/pgbouncer-deployment.yaml
Comment (jedcunningham) [inline chart/templates/pgbouncer/pgbouncer-deployment.yaml]:
```suggestion
      {{- if .Values.pgbouncer.extraAnnotations }}
        {{- toYaml .Values.pgbouncer.extraAnnotations | nindent 8 }}
      {{- end }}
```

Let's indent it like this instead.

**Review by jedcunningham — CHANGES_REQUESTED** [review]

File: chart/templates/pgbouncer/pgbouncer-deployment.yaml
Comment (jedcunningham) [reply chart/templates/pgbouncer/pgbouncer-deployment.yaml]: Just to add a little more color here, there is nothing wrong with how you have it, we just had a few other PRs recently that did some formatting cleanup and I'd like to keep things consistent 👍

**Review by jedcunningham — APPROVED** [review]

**Comment by jedcunningham** [comment]
Thanks @dim-ops! Congrats on your first commit 🎉

**Pull request merged**
